In [ ]:
setwd("C:/Users/diogo/code/Analyse de Donne/Train for the Test")

In [1]:
data(iris)
iris

Sepal.Length Sepal.Width Petal.Length Petal.Width Species  
1   5.1          3.5         1.4          0.2         setosa   
2   4.9          3.0         1.4          0.2         setosa   
3   4.7          3.2         1.3          0.2         setosa   
4   4.6          3.1         1.5          0.2         setosa   
5   5.0          3.6         1.4          0.2         setosa   
6   5.4          3.9         1.7          0.4         setosa   
7   4.6          3.4         1.4          0.3         setosa   
8   5.0          3.4         1.5          0.2         setosa   
9   4.4          2.9         1.4          0.2         setosa   
10  4.9          3.1         1.5          0.1         setosa   
11  5.4          3.7         1.5          0.2         setosa   
12  4.8          3.4         1.6          0.2         setosa   
13  4.8          3.0         1.4          0.1         setosa   
14  4.3          3.0         1.1          0.1         setosa   
15  5.8          4.0         1.2          0.2         setosa   
16  5.7          4.4         1.5          0.4         setosa   
17  5.4          3.9         1.3          0.4         setosa   
18  5.1          3.5         1.4          0.3         setosa   
19  5.7          3.8         1.7          0.3         setosa   
20  5.1          3.8         1.5          0.3         setosa   
21  5.4          3.4         1.7          0.2         setosa   
22  5.1          3.7         1.5          0.4         setosa   
23  4.6          3.6         1.0          0.2         setosa   
24  5.1          3.3         1.7          0.5         setosa   
25  4.8          3.4         1.9          0.2         setosa   
26  5.0          3.0         1.6          0.2         setosa   
27  5.0          3.4         1.6          0.4         setosa   
28  5.2          3.5         1.5          0.2         setosa   
29  5.2          3.4         1.4          0.2         setosa   
30  4.7          3.2         1.6          0.2         setosa   
⋮   ⋮            ⋮           ⋮            ⋮           ⋮        
121 6.9          3.2         5.7          2.3         virginica
122 5.6          2.8         4.9          2.0         virginica
123 7.7          2.8         6.7          2.0         virginica
124 6.3          2.7         4.9          1.8         virginica
125 6.7          3.3         5.7          2.1         virginica
126 7.2          3.2         6.0          1.8         virginica
127 6.2          2.8         4.8          1.8         virginica
128 6.1          3.0         4.9          1.8         virginica
129 6.4          2.8         5.6          2.1         virginica
130 7.2          3.0         5.8          1.6         virginica
131 7.4          2.8         6.1          1.9         virginica
132 7.9          3.8         6.4          2.0         virginica
133 6.4          2.8         5.6          2.2         virginica
134 6.3          2.8         5.1          1.5         virginica
135 6.1          2.6         5.6          1.4         virginica
136 7.7          3.0         6.1          2.3         virginica
137 6.3          3.4         5.6          2.4         virginica
138 6.4          3.1         5.5          1.8         virginica
139 6.0          3.0         4.8          1.8         virginica
140 6.9          3.1         5.4          2.1         virginica
141 6.7          3.1         5.6          2.4         virginica
142 6.9          3.1         5.1          2.3         virginica
143 5.8          2.7         5.1          1.9         virginica
144 6.8          3.2         5.9          2.3         virginica
145 6.7          3.3         5.7          2.5         virginica
146 6.7          3.0         5.2          2.3         virginica
147 6.3          2.5         5.0          1.9         virginica
148 6.5          3.0         5.2          2.0         virginica
149 6.2          3.4         5.4          2.3         virginica
150 5.9          3.0         5.1          1.8         virginica

In [5]:
summary(iris)

  Sepal.Length    Sepal.Width     Petal.Length    Petal.Width   
 Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  
 1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  
 Median :5.800   Median :3.000   Median :4.350   Median :1.300  
 Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  
 3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  
 Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  
       Species  
 setosa    :50  
 versicolor:50  
 virginica :50  
                
                
                

Corelação de Person

In [2]:
cor_test_sepal <- cor.test(iris$Sepal.Length, iris$Sepal.Width)
cor_test_sepal


	Pearson's product-moment correlation

data:  iris$Sepal.Length and iris$Sepal.Width
t = -1.4403, df = 148, p-value = 0.1519
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.27269325  0.04351158
sample estimates:
       cor 
-0.1175698 


O valor de p-value (0.1519) é maior que 0.05, o que significa que não há evidência suficiente para rejeitar a hipótese nula (H0). A hipótese nula nesse caso afirma que não há correlação entre Sepal.Length e Sepal.Width.

In [4]:
cor_test_sepal <- cor.test(iris$Petal.Length, iris$Petal.Width)
cor_test_sepal


	Pearson's product-moment correlation

data:  iris$Petal.Length and iris$Petal.Width
t = 43.387, df = 148, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9490525 0.9729853
sample estimates:
      cor 
0.9628654 


Há uma correlação positiva muito forte entre o comprimento e a largura das pétalas, o que faz sentido do ponto de vista biológico, já que essas dimensões provavelmente crescem juntas em muitas espécies de flores, incluindo as do dataset iris.

# 3. Análise em Componentes Principais (ACP)

Agora, realizamos uma ACP (Análise em Componentes Principais) com as variáveis adequadas (normalmente as quantitativas). Vamos eliminar a variável Species para focar nas variáveis numéricas.

In [7]:
library(FactoMineR)

In [8]:
iris_data <- iris[, 1:4] # Selecionar as variáveis quantitativas
pca_result <- PCA(iris_data, scale.unit = TRUE)

summary(pca_result)


Call:
PCA(X = iris_data, scale.unit = TRUE) 


Eigenvalues
                       Dim.1   Dim.2   Dim.3   Dim.4
Variance               2.918   0.914   0.147   0.021
% of var.             72.962  22.851   3.669   0.518
Cumulative % of var.  72.962  95.813  99.482 100.000

Individuals (the 10 first)
                 Dist    Dim.1    ctr   cos2    Dim.2    ctr   cos2    Dim.3
1            |  2.319 | -2.265  1.172  0.954 |  0.480  0.168  0.043 | -0.128
2            |  2.202 | -2.081  0.989  0.893 | -0.674  0.331  0.094 | -0.235
3            |  2.389 | -2.364  1.277  0.979 | -0.342  0.085  0.020 |  0.044
4            |  2.378 | -2.299  1.208  0.935 | -0.597  0.260  0.063 |  0.091
5            |  2.476 | -2.390  1.305  0.932 |  0.647  0.305  0.068 |  0.016
6            |  2.555 | -2.076  0.984  0.660 |  1.489  1.617  0.340 |  0.027
7            |  2.468 | -2.444  1.364  0.981 |  0.048  0.002  0.000 |  0.335
8            |  2.246 | -2.233  1.139  0.988 |  0.223  0.036  0.010 | -0.089
9      